pyproject.toml

```diff
@@ -39,10 +39,12 @@ include = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.9,<3.11"
-
+python = "^3.10,<4"
 h5py = "^3.2.1"
-numpy = "!=1.19.4, ~1.23.5"
+numpy = [
+    {version="!=1.19.4, ^1.23.5", python= "<3.12"},
+    {version="^1.26.0", python= ">=3.12"}
```

Comment:
numpy 1.26 seems to support Python versions from 3.9 to 3.12, so the same version could be used no matter the version of Python

(from [numpy-1-26-0 release notes](https://numpy.org/doc/1.26/release/1.26.0-notes.html#numpy-1-26-0-release-notes))